easybuild/easyconfigs/e/EGSnrc/EGSnrc-2019a.eb

```diff
@@ -0,0 +1,94 @@
+easyblock = 'Tarball'
+name = 'EGSnrc'
+version = '2019a'
+
+homepage = 'http://nrc-cnrc.github.io/EGSnrc/'
+description = """
+EGSnrc is a software toolkit to perform Monte Carlo simulation of ionizing radiation transport through matter.\
+It models the propagation of photons, electrons and positrons with kinetic energies\
+between 1 keV and 10 GeV, in homogeneous materials."""
+
+source_urls = ['https://github.com/nrc-cnrc/EGSnrc/archive/']
+sources = ['v%(version)s.tar.gz']
+patches = ['log-dir_finanlize.patch']
+checksums = [
+    'd6f9198aa2d27645134cb5be477165547df6a66b2c0eed11e73c7d118ea312a8',  # v2019a.tar.gz
+    'd54989aead38e7112d327a7ff299587c99c9d2b5c5df981322ebf1d354df34f5',  # log-dir_finanlize.patch
+]
+
+toolchain = {'name': 'system', 'version': ''}
```

Comment:
It should be
toolchain = SYSTEM
if you want to use it.

Reply:
When using GCCcore/6.4.0 or GCC 6.4.0-2.28, the compilation process fails when trying to compile pegs4. I don't know wether it's some flags that are set differently when using a GCC toolchain, but I couldn't avoid it any other way.
But I will gladly change 'system' to 'SYSTEM'

Reply:
Quick revision: I can't change to SYSTEM, easybuild won't accept it. We still have Easybuild/3.9.4 installed and we need to keep it that way for some time until changing to 4.X

Reply:
There is no reason to hold back on upgrading to a newer easybuild release. You can just choose which ever version you want through the module system...

Reply:
And if using toolchain GCC/GCCcore causes problems, then you should figure out why exactly and we can probably show you how to fix that.

Reply:
Meanwhile I believe that it is the different FLAGS that cause the problems. Basically, the whole installation happens within this script: ./HEN_HOUSE/scripts/configure

At some point the flags are displayed and then the compilation begins.  Here the logs differ: with SYSTEM / dummy only fPIC is set. With GCC as toolchain the following flags are set additionally:
FCFLAGS = -O2 -ftree-vectorize -march=native -fno-math-errno -fPIC

I have the feeling that for some unknown reason this causes the compilation of pegs4.

Can I change the FCFLAGS somehow without having to touch the Easyblock?

Reply:
Yes, you can start with
toolchainopts = {'optarch': False, 'noopt': True, 'pic': True}

That should be a good starting point, and then you can change those to see which causes the actual problem.

You could also pick out the actual compile error and put it in a gist for us to look at.

Reply:
optarch = False seems to apply, since now the FCFLAGS are set to -O0, but unfortunately noopt: False does not seem to apply.

You can dig into the log here, if you wanthttps://gist.github.com/HPC-UniOldenburg/b5ce1a1c4c7d8b6866dddb2e7093d8e9#file-easybuild-egsnrc-log

And thank you very much for your help so far, @akesandgren !

Reply:
optarch false turns off -march=native only, noopt changes from -O2 -ftree-vectorize  to -O0
So it does take effect.

Reply:
Looks like we'd need the configure.log to see what's actually going wrong.

So eb ... --disable-cleanup-builddir and put that configure.log it talks about in a new gist.